src/controllers/userController.js

```diff
@@ -0,0 +1,184 @@
+    return res.status(201).json({ message: "User created", newUser, token });
+  } catch (error) {
+    return res.status(500).json({ message: error.message });
+  }
+};
+
+// verifying Email
+
+const verifyEmail = async (req, res, next) => {
+  try {
+    const { token } = req.query;
+    const decodedToken = await jwt.verify(token, process.env.SECRET);
+    const user = await db.execute("SELECT * FROM users WHERE email = ?", [
+      {
+        email: decodedToken.email,
+      },
+    ]);
+
+    if (user.is_verified) {
+      return successResMsg(res, 200, {
+        message: "user verified already",
+      });
+    }
+
+    const verify = await db.execute(
+      "UPDATE users SET is_verified = true WHERE is_verified = false"
+    );
+    return res
+      .status(200)
+      .json({ message: "User verified successfully" });
+  } catch (error) {
+    return res.status(500).json({ message: error.message });
+  }
+};
+// logging in a user
+const login = async (req, res, next) => {
+  try {
+    const { email, password } = req.body;
+    // validate with joi
+    await UserLogin.validateAsync(req.body);
+    //  checking email and password match
+    if (email && password) {
+      const user = await db.execute("SELECT * FROM users WHERE email =?", [
+      email,
+      ]);
+      if (!user) {
+        return res.status(400).json({
+          message: "email address not found.",
+        });
+      }
+      const passMatch = await bcrypt.compare(password, user.password);
+      if (!passMatch) {
+        return res.status(400).json({ message: "incorrect details" });
+      }
+      if (!user.is_verified) {
+        return res.status(400).json({
+          message: "Unverified account.",
+        });
+      }
+    }
+    // creating a payload
+    const payload = {
+      id: email.id,
+      email: email.email,
+      role: email.role,
+    };
+
+    const token = await jwt.sign(payload, process.env.SECRET_TOKEN, {
+      expiresIn: "1h",
+    });
+    return res.status(200).json({
+      message: "User logged in sucessfully",
+      token,
+    });
+  } catch (error) {
+    return res.status(500).json,({ message: error.message });
+  }
+};
+
+
+const getAllUsers = async (req, res, next) => {
+  try {
+    const users = await db.query(
+      `SELECT * FROM users `
+    );
+    if (
+  !users
+    ) {
+      return res.status(404).json({
+        message: "Users not found",
+      });
+    }
+   
+    return res.status(200).json({
+      message: "Users fetch successfully",
+     users
+  
+    });
+  } catch (error) {
+    return res.status(500).json({ message: error.message });
+  }
+};
+
+
+const getAUser = async (req, res, next) => {
+  try {
+    const { id } = req.params;
+    const user = await db.query("SELECT * FROM users WHERE id =?", [id]);
+
+    if (!user) {
+      return res.status(404).json({
+        message: "User not found",
+      });
+    }
+
+    return res.status(200).json({
+      message: "Users fetch successfully",
```

Comment:
instead of just returning the message, don't you think we should return the user data 
```
res.status(200).json({
        data: user.row[0]
)};
```
but we only like want to send all the user data, probably just id, name, and phone-number

Reply:
please check the import thing, i have not even tested any of the endpoint